terraform/shared/shared.tf.json.template.py

```diff
@@ -41,6 +41,15 @@
                 }
             }
         },
+        'aws_s3_bucket_logging': {
+            'shared_cloudtrail': {
+                'bucket': '${aws_s3_bucket.shared_cloudtrail.id}',
+                'target_bucket': '${aws_s3_bucket.logs.id}',
```

Comment:
Source and target bucket must be in the same region. In some accounts, the trail bucket is in a different region. 

https://docs.aws.amazon.com/AmazonS3/latest/userguide/ServerLogs.html

Please check how Prowler checks for this. Maybe we could use the https://docs.aws.amazon.com/AmazonS3/latest/userguide/cloudtrail-logging.html option so that access to the trail bucket is logged in CloudTrail.

Alternatively, I'd be willing to discontinue the old trails and set up new ones in the "right" region. Please determine in which accounts the trail region is different.

Reply:
> Please check how Prowler checks for this. Maybe we could use the https://docs.aws.amazon.com/AmazonS3/latest/userguide/cloudtrail-logging.html option so that access to the trail bucket is logged in CloudTrail.

Prolwer uses `aws cloudtrail describe-trails` to find the Cloudtrail output bucket, and then `aws s3api get-bucket-logging` to see if logging is enabled on the bucket. https://github.com/mberger/aws-cis-security-benchmark/blob/master/checks/check26

```
$ aws s3api get-bucket-logging --bucket edu-ucsc-gi-singlecell-azul-storage-daniel --query 'LoggingEnabled.TargetBucket' --output text
edu-ucsc-gi-platform-hca-dev-logs.us-east-1
```

Based on this it is the "Server access logging" setting that Prowler is looking for, and using "AWS CloudTrail data events" on the bucket won't suffice.

![Screen Shot 2022-12-12 at 1 36 06 PM](https://user-images.githubusercontent.com/46407464/207159708-c6776d2a-a188-476b-a95f-3f66b8fb633f.png)

Reply:
> Alternatively, I'd be willing to discontinue the old trails and set up new ones in the "right" region. Please determine in which accounts the trail region is different.

```
anvilbox
AWS_DEFAULT_REGION  us-east-1
cloudtrail_bucket   us-east-1
cloudtrail_trail    us-east-1

anvildev
AWS_DEFAULT_REGION  us-east-1
cloudtrail_bucket   us-east-1
cloudtrail_trail    us-east-1

anvilprod
AWS_DEFAULT_REGION  us-east-1
cloudtrail_bucket   us-east-1
cloudtrail_trail    us-east-1

dev
AWS_DEFAULT_REGION  us-east-1
cloudtrail_bucket   us-west-2  <-- bucket outside default region
cloudtrail_trail    us-west-2  <-- trail outside default region

prod
AWS_DEFAULT_REGION  us-east-1
cloudtrail_bucket   us-east-1
cloudtrail_trail    us-west-2  <-- trail outside default region

sandbox
AWS_DEFAULT_REGION  us-east-1
cloudtrail_bucket   us-east-1
cloudtrail_trail    us-east-1
```